app/agent/utils.ts

```diff
@@ -111,10 +112,32 @@ export const withSsh = async <T>(
     15,
     500,
   );
+
+  const context = getActivityContext();
+  let finish: () => void = () => {};
+  const finished = new Promise<void>((resolve) => {
+    finish = resolve;
+  });
+
+  if (context != null) {
+    Promise.race([context.cancelled, finished]).catch(() => {
```

Comment:
Why is this promise floating here?

Reply:
It's the most convenient way to handle cancellation. Even though it's floating, it gets resolved quickly after the function finishes, so it shouldn't be a problem.